Sprint 5 SonarQube Refactoring

JayChouteau opened the pull request:
This pull request includes all refactoring done based on the sonarQube analysis on the master branch at the end of sprint 5. It includes bugs, vulnerabilities, code smells, and dupplication. Most of them were fixed. A few of them were resolved as won't fix since the code it was located in was currently being refactored or modified in different issues. The acceptance criteria of the linked issue were fulfilled. 

This pull request should be merged after (on top) of all the current pull-requests so all the little refactoring changes apply on top of the develop branch. All information pertinent to the SonarQube is available on Google Drive in quality metrics. The description of issues and resolution of those will have to show in the release 2 documentation as well.

JayChouteau committed:
SonarQube Refactoring (#124)
- Bug Fixing (all)
- Vulnerability issue fixing (all)

JayChouteau committed:
SonarQube Refactoring (#124)
- code smells (30 left)

JayChouteau committed:
SonarQube Refactoring (#124)
- reverting back a change

JayChouteau committed:
SonarQube Refactoring (#124)
- Code smells resolved (all)

JayChouteau committed:
SonarQube Refactoring (#124)
- Deleted duplicate dependencies in pom file
- Moved the e2e testing folder

jeffbuonamici commented on Server/src/main/java/local/DatabaseConnection.java:
```diff
@@ -9,7 +9,7 @@
     //Credentials should be set to that which you are using on your local DB server.
     private static final String USER = "root"; // todo  use username and password specific to your machine
-    private static final String PASSWORD = "";
+    private static final String PASSWORD = "root";
```
Just remove this.

JayChouteau replied:
Hey Jeff, it is now resolved!

jeffbuonamici reviewed (changes requested):
Thanks for the code clean up. There's just one small change.

JayChouteau committed:
SonarQube Refactoring (#124)
- Took out some unused code (methods and classes)
- Didn't touch on the UI controllers or the datapreprocessing parts since they are being refactored/worked on at the moment

JayChouteau commented:
In the last commit, I removed some methods and files that were unused in our application. I left the placeholder for the feature to come (threshold and recommendation) and didn't remove anything from the controller part and data-preprocessing since there was a lot of work that was done on those that are not yet merged into our develop branch.

PaulMicu commented:
can you please put a screenshot of sonarqube and clover after your changes

PaulMicu reviewed (approved):
some of these changes seem silly but if that's what sonarqube wants i guess it's ok

jeffbuonamici reviewed (approved):
Looks good now.
